zipline/finance/blotter.py

```diff
@@ -162,7 +162,8 @@ def execute_cancel_policy(self, event):
             warn = self.cancel_policy.warn_on_cancel
 
             for orders_by_sid in itervalues(self.open_orders):
-                for order in orders_by_sid:
+                # So we don't mutate the list as we cancel orders
+                for order in orders_by_sid[:]:
```

Comment:
The function directly above this is `cancel_all_orders_for_asset`, which handles this correctly (and documents that this is an issue.)  Is there a reason we're not just using that here?